src/libs/actions/IOU.js

```diff
@@ -720,11 +721,267 @@ function payIOUReport({
+        newSequenceNumber,
+        CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        iouReport.total,
+        iouReport.currency,
+        '',
+        [recipient],
+        paymentMethodType,
+        '',
+        iouReport.reportID,
+    );
+
+    // First, add data that will be used in all cases
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${chatReport.reportID}`,
+            value: {
+                ...chatReport,
+                lastVisitedTimestamp: Date.now(),
+                lastReadSequenceNumber: newSequenceNumber,
+                maxSequenceNumber: newSequenceNumber,
+                lastMessageText: optimisticPaidReportAction.message[0].text,
+                lastMessageHtml: optimisticPaidReportAction.message[0].html,
+                hasOutstandingIOU: false,
+                iouReportID: iouReport.reportID,
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
+            value: {
+                [optimisticPaidReportAction.sequenceNumber]: {
+                    ...optimisticPaidReportAction,
+                    pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                },
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReport.reportID}`,
+            value: iouReport,
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
+            value: {
+                [optimisticPaidReportAction.sequenceNumber]: {
+                    pendingAction: null,
+                },
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
+            value: {
+                [optimisticPaidReportAction.sequenceNumber]: {
+                    ...optimisticPaidReportAction,
+                    pendingAction: null,
+                },
+            },
+        },
+    ];
+
+    API.write('PayMoneyRequest', {
+        iouReportID: iouReport.reportID,
+        paidReportActionID: optimisticPaidReportAction.reportActionID,
+        paymentMethodType,
+        clientID: optimisticPaidReportAction.sequenceNumber,
+    }, {optimisticData, successData, failureData});
+}
+
+function payMoneyRequestElsewhere(chatReport, iouReport, recipient) {
+    payMoneyRequest(chatReport, iouReport, recipient, CONST.IOU.PAYMENT_TYPE.ELSEWHERE);
+}
+
+function payMoneyRequestViaPaypal(chatReport, iouReport, recipient) {
+    payMoneyRequest(chatReport, iouReport, recipient, CONST.IOU.PAYMENT_TYPE.PAYPAL_ME);
```

Comment:
I guess lets add docs here too so it is same as the other methods.

Reply:
Yeah sorry forgot to add them in this commit.

Reply:
np!